core/common/src/internal/format/parser/ParserOperation.kt

```diff
@@ -50,7 +50,12 @@ internal class NumberSpanParserOperation<Output>(
 
     init {
         require(consumers.all { (it.length ?: Int.MAX_VALUE) > 0 })
-        require(consumers.count { it.length == null } <= 1)
+        require(consumers.count { it.length == null } <= 1) {
+            val fieldNames = consumers.filter { it.length == null }.map { it.whatThisExpects }
+            "At most one variable-length numeric field in a row is allowed, but got several: $fieldNames. " +
```

Comment:
WDYT about adding a recommendation how to avoid the error here?
E.g. `Consider adding a `char(' ') separator between ...`

Reply:
If you're an absolute novice and are just trying this functionality out, I think you're extremely unlikely to encounter an exception like this. You need to either explicitly unset the padding or to write a format that contains both the year and the fractional part of the second without any separators.